### DISCUSSION TITLE
WHIP and round trip time

### DISCUSSION THREAD

**Discussion by yinshanyang** [obsproject/obs-studio#9967] [General]
I’m trying to calculate the round trip time delivered by OBS through WHIP, but it seems like `RtcpSrReporter` does not send the `LSR` and `DLSR` by default.

Or am I mistaken? Or if someone could kindly point me in the right direction.

**Comment by Fenrirthviti** [comment]
As a note, it might be best to join the WHIP/WebRTC discussion Discord server for a more active response: https://discord.com/invite/BQT5Mdx52r

EDIT: Fixed link, it was apparently a single-use link on accident.

**Comment by Sean-Der** [comment]
Hey @yinshanyang were you able to get an answer for this? I am happy to go work on libdatachannel if this is still a problem.

@Fenrirthviti feel free to tag me on any issues, I don't mind answering on GitHub also!

**Reply by yinshanyang** [reply]
Hey @Sean-Der, thanks for the response.

> I am happy to go work on libdatachannel if this is still a problem.

If that’s possible, I think it’ll be useful. My use case for it is to obtain the RTT across multiple streams so as to crudely synchronize them together. But I also believe it will be useful in the context of observing stream quality.

At the moment, my workaround is to get RTT through the ICE binding request instead, but it would be nice to have the option of getting it through the RTCP reports instead.
